sw/otbn/crypto/tests/mlkem/mlkem_base_decap_test.s

```diff
@@ -0,0 +1,3298 @@
+/* Copyright "Towards ML-KEM & ML-DSA on OpenTitan" Authors */
+/* Licensed under the Apache License, Version 2.0, see LICENSE for details. */
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Modified by Ruben Niederhagen and Hoang Nguyen Hien Pham - authors of */
+/* "Improving ML-KEM & ML-DSA on OpenTitan - Efficient Multiplication Vector Instructions for OTBN" */
+/* (https://eprint.iacr.org/2025/2028) */
+/* Copyright Ruben Niederhagen and Hoang Nguyen Hien Pham. */
+
+
+/*
+ * Testwrapper for mlkem_decap
+*/
+
+.section .text.start
+
+#define STACK_SIZE 20000
+#define CRYPTO_BYTES 32
+
+#if KYBER_K == 2
+  #define CRYPTO_PUBLICKEYBYTES  800
+  #define CRYPTO_SECRETKEYBYTES  1632
+  #define CRYPTO_CIPHERTEXTBYTES 768
+#elif KYBER_K == 3
+  #define CRYPTO_PUBLICKEYBYTES  1184
+  #define CRYPTO_SECRETKEYBYTES  2400
+  #define CRYPTO_CIPHERTEXTBYTES 1088
+#elif KYBER_K == 4
+  #define CRYPTO_PUBLICKEYBYTES  1568
+  #define CRYPTO_SECRETKEYBYTES  3168
+  #define CRYPTO_CIPHERTEXTBYTES 1568
+#endif
+
+/* Entry point. */
+.globl main
+main:
+  /* Init all-zero register. */
+#ifdef RTL_ISS_TEST
```

Comment:
nit: it might make sense to put this block as a subroutine in a file called e.g. `test_preamble.s` that gets imported and then called by every test so it's easier to change. But also, I'm a bit confused: why isn't the mismatch with non-cleared registers a problem for tests that aren't ML-KEM?